Index: b/src/driver.h
===================================================================
--- a/src/driver.h
+++ b/src/driver.h
@@ -17,7 +17,7 @@
 #include "timing.h"
 #include "accel.h"
 
-#define MAX_REGS 5000 /* VESA needs a lot of storage space */
+#define MAX_REGS 11196 /* VESA needs a lot of storage space */
 
 typedef struct {
     void (*savepalette)(unsigned char *red, unsigned char *green, unsigned char *blue);
