Index: b/src/config/libvga.config
===================================================================
--- a/src/config/libvga.config
+++ b/src/config/libvga.config
@@ -31,7 +31,15 @@
 # mouse DRMOUSE4DS	# Digital Research double-wheeled mouse
 # mouse none		# None
 
-mouse Microsoft
+mouse unconfigured
+
+# (DEBIAN NOTE: the mouse used to default to microsoft, but this was changed
+#  to fix bug #13458. If your mouse used to work fine, you can simply change
+#  it back to read "microsoft" again. If you are careful to change *only that
+#  one word*, and not to add or remove extra whitespace, the package
+#  installation will continue to update this file without requiring user
+#  intervention because of a modified config file.
+#  This applies to all mouse types, not just microsoft.)
 
 # Mouse/keyboard customisation by 101 (Attila Lendvai). If you have any good
 # ideas you can reach me at 101@kempelen.inf.bme.hu
Index: b/src/mouse/mouse.c
===================================================================
--- a/src/mouse/mouse.c
+++ b/src/mouse/mouse.c
@@ -47,6 +47,13 @@ int mouse_init_return_fd(char *dev, int 
 	    currentinthandler = NULL;
 	
 	    /* Initialize mouse device. */
+	    if (m_type == MOUSE_UNCONFIGURED) {
+	      fprintf(stderr,
+		      "Warning: you have not yet configured your mouse type. If you have no mouse,\n"
+		      "setting the type to `none' in " SVGALIB_CONFIG_FILE " will get rid of this\n"
+		      "annoying message.\n");
+	      return -1;
+	    }
 	    if (m_type == MOUSE_NONE || m_type < MOUSE_MICROSOFT || m_type > MOUSE_LAST)
 		return -1;
 	    if (ms_init())
Index: b/src/mouse/vgamouse.h
===================================================================
--- a/src/mouse/vgamouse.h
+++ b/src/mouse/vgamouse.h
@@ -24,7 +24,8 @@ extern "C"
 #define MOUSE_PNP 12
 #define MOUSE_WACOM_GRAPHIRE 13
 #define MOUSE_DRMOUSE4DS 14
-#define MOUSE_LAST MOUSE_DRMOUSE4DS
+#define MOUSE_UNCONFIGURED 15
+#define MOUSE_LAST MOUSE_UNCONFIGURED
 
 #define MOUSE_ORIENTATION_VERTICAL 0
 #define MOUSE_ORIENTATION_HORIZONTAL 1
Index: b/src/vga.c
===================================================================
--- a/src/vga.c
+++ b/src/vga.c
@@ -3589,7 +3589,7 @@ static char *conf_mousenames[] =
 {
   "Microsoft", "MouseSystems", "MMSeries", "Logitech", "Busmouse", "PS2",
     "MouseMan", "gpm", "Spaceball", "none", "IntelliMouse", "IMPS2", "pnp", 
-    "WacomGraphire", "DRMOUSE4DS", NULL};
+    "WacomGraphire", "DRMOUSE4DS", "unconfigured", NULL};
 
 static int check_digit(char *ptr, char *digits)
 {
