Index: b/src/vga.c
===================================================================
--- a/src/vga.c
+++ b/src/vga.c
@@ -38,6 +38,7 @@
 #include "vgaregs.h"
 #include "vgaversion.h"
 
+#include <syslog.h>
 #ifdef BACKGROUND
 #include "vgabg.h"
 
@@ -2519,6 +2520,12 @@ void vga_gettextfont(void *font)
 {
     unsigned int getsize;
 
+    /* robert@debian.org, May, 26th 2002: check for valid font_buf buffer */
+    if (!font_buf1) {
+	 syslog(LOG_DEBUG, "svgalib: uninitialized variable: font_buf1");
+	 return;
+    }	 
+	    
     getsize = fontbufsize;
     if (getsize > FONT_SIZE)
 	getsize = FONT_SIZE;
@@ -2530,6 +2537,12 @@ void vga_gettextfont(void *font)
 void vga_puttextfont(void *font)
 {
     unsigned int putsize;
+    
+    /* robert@debian.org, May, 26th 2002: check for valid font_buf buffer */
+    if (!font_buf1 || !font_buf2) {
+	 syslog(LOG_DEBUG, "svgalib: uninitialized variable: font_buf1 or font_buf2");
+	 return;
+    }	 
 
 #ifdef BACKGROUND
         if (-1 == mprotect(font_buf1,FONT_SIZE*2,PROT_READ|PROT_WRITE))
