Index: b/gl/Makefile
===================================================================
--- a/gl/Makefile
+++ b/gl/Makefile
@@ -32,7 +32,7 @@ all:	libvgagl.a
 
 libvgagl.so.$(VERSION): $(MODULES)
 	$(CC) -s -shared -Wl,-soname,libvgagl.so.$(MAJOR_VER) -o libvgagl.so.$(VERSION) \
-	  $(MODULES) $(SHLIBLIBS)
+	  $(MODULES) $(SHLIBLIBS) -L$(srcdir)/sharedlib -lvga
 
 libvgagl.a: $(MODULES)
 	rm -f libvgagl.a
