Index: b/gl/Makefile
===================================================================
--- a/gl/Makefile
+++ b/gl/Makefile
@@ -30,9 +30,12 @@ MODULES	 = grlib.o driver.o line.o palet
 all:	libvgagl.a
 .PHONY: all clean dep
 
+libvgagl.so: libvgagl.so.$(VERSION)
 libvgagl.so.$(VERSION): $(MODULES)
 	$(CC) -s -shared -Wl,-soname,libvgagl.so.$(MAJOR_VER) -o libvgagl.so.$(VERSION) \
 	  $(MODULES) $(SHLIBLIBS) -L$(srcdir)/sharedlib -lvga
+	ln -sf libvgagl.so.$(VERSION) libvgagl.so
+	ln -sf libvgagl.so.$(VERSION) libvgagl.so.$(MAJOR_VER)
 
 libvgagl.a: $(MODULES)
 	rm -f libvgagl.a
@@ -65,7 +68,7 @@ dep:
 	$(CC) $(INCLUDES) -MM $(patsubst %.o,$(srcdir)/gl/%.c,$(MODULES)) >>.depend.gl
 
 clean:
-	rm -f .depend.gl *.bak *.o *~ libvgagl.a libvgagl.so.$(VERSION)
+	rm -f .depend.gl *.bak *.o *~ libvgagl.a libvgagl.so*
 
 #
 # include a dependency file if one exists
Index: b/Makefile
===================================================================
--- a/Makefile
+++ b/Makefile
@@ -135,8 +135,8 @@ endif
 		$(addprefix $(sharedlibdir)/,$(notdir $(SHAREDLIBS))).
 	@for foo in $(notdir $(SHAREDLIBS)); do \
 		$(INSTALL_SHLIB) sharedlib/$$foo $(sharedlibdir)/$$foo; \
-		(cd $(sharedlibdir); \
-		 ln -sf $$foo `echo $$foo | sed 's/\.so\..*/.so/'` ); \
+		cp -d sharedlib/`echo $$foo | sed 's/\.so\..*/.so/'` $(sharedlibdir); \
+		cp -d sharedlib/`echo $$foo | sed 's/\.so\..*/.so/'`.$(MAJOR_VER) $(sharedlibdir); \
 	done
 
 ifdef INSTALLAOUTDIR
@@ -150,8 +150,8 @@ installaoutcompat:
 		$(addprefix $(INSTALLAOUTDIR),$(notdir $(SHAREDLIBS))).
 	@for foo in $(notdir $(SHAREDAOUTLIBS)); do \
 		$(INSTALL_SHLIB) sharedlib/$$foo $(TOPDIR)/$(INSTALLAOUTDIR)$$foo; \
-		(cd $(TOPDIR)/$(INSTALLAOUTDIR); \
-		 ln -sf $$foo `echo $$foo | sed 's/\.so\..*/.so/'` ); \
+		cp -d sharedlib/`echo $$foo | sed 's/\.so\..*/.so/'` $(sharedlibdir); \
+		cp -d sharedlib/`echo $$foo | sed 's/\.so\..*/.so/'`.$(MAJOR_VER) $(sharedlibdir); \
 	done
 
 else
Index: b/src/Makefile
===================================================================
--- a/src/Makefile
+++ b/src/Makefile
@@ -389,11 +389,15 @@ all:	libvga.so.$(VERSION)
 libvga.so.$(VERSION): $(ALLOBJS)
 	$(CC) -s -shared -Wl,-soname,libvga.so.$(MAJOR_VER) \
 	  -o libvga.so.$(VERSION) $(ALLOBJS) $(SHLIBLIBS) -lm
+	ln -sf libvga.so.$(VERSION) libvga.so
+	ln -sf libvga.so.$(VERSION) libvga.so.$(MAJOR_VER)
 
 $(sharedlibdir)/libvga.so.$(VERSION): libvga.so.$(VERSION)
 	$(INSTALL_PROGRAM) $< $(sharedlibdir)/$<
 
 install: $(sharedlibdir)/libvga.so.$(VERSION)
+	cp -df sharedlib/libvga.so $(sharedlibdir)/
+	cp -df sharedlib/libvga.so.$(MAJOR_VER) $(sharedlibdir)/
 
 endif	  
 
