Author: Guillem Jover <guillem@debian.org>

--- svgalib-1.9.18.orig/src/Makefile	2004-01-21 09:44:51.000000000 +0100
+++ svgalib-1.9.18/src/Makefile	2004-01-21 09:50:08.000000000 +0100
@@ -118,7 +118,7 @@
 endif
 ifdef INCLUDE_EGA_DRIVER
   VGA_DEFINES += -DINCLUDE_EGA_DRIVER
-  DRIVERS += drivers/egadrv.o
+  DRIVERS += egadrv.o
   ifdef INCLUDE_EGA_DRIVER_TEST
     VGA_DEFINES += -DINCLUDE_EGA_DRIVER_TEST
   endif
