Author: Osvaldo La Rosa <info@brlspeak.net>

Index: b/src/vga.c
===================================================================
--- a/src/vga.c
+++ b/src/vga.c
@@ -4489,6 +4489,7 @@ if(B8000_MEM_POINTER==NULL){
        presumably not the idea if it's been explicitly closed here! */
     __svgalib_mem_fd=-2;
    
+    if (mouse_type != MOUSE_NONE) {
 #ifdef DEBUG
 	    printf("svgalib: Opening mouse (type = %x).\n", mouse_type | mouse_modem_ctl);
 #endif
@@ -4496,6 +4497,7 @@ if(B8000_MEM_POINTER==NULL){
 		printf("svgalib: Failed to initialize mouse.\n");
 	    else
 		mouse_open = 1;
+    }
 
     /* Michael: I assume this is a misunderstanding, when svgalib was developed,
        there were no saved uids, thus setting effective uid sufficed... */
