Author: Guillem Jover <guillem@debian.org>

---
 gl/inlstring.h |    2 +-
 gl/mem.S       |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

--- a/gl/inlstring.h
+++ b/gl/inlstring.h
@@ -258,7 +258,7 @@ static inline void *
 	       "cmpl $0,%%edx\n\t"	/* none left? */
 	       "jle 4f\n\t"	/* finished */
 
-	       "mov %%ecx,%%eax\n\t"
+	       "movl %%ecx,%%eax\n\t"
 	       "shrl $8,%%ecx\n\t"	/* R in cl */
 
 	       "3:\n\t"		/* write last few pixels */
--- a/gl/mem.S
+++ b/gl/mem.S
@@ -76,7 +76,7 @@ ENTRY(memcpy4to3)
 	jz 4f			/* none left */
 
 3:	movl (%esi),%eax
-	movw %eax,(%edi)
+	movw %ax,(%edi)
 	shrl $16,%eax
 	movb %al,2(%edi)
 	addl $4,%esi
