Author: Guillem Jover <guillem@debian.org>

---
 utils/gtf/gtfcalc.c |    4 ++++
 utils/gtf/scitech.h |   12 ++++++++++++
 2 files changed, 16 insertions(+)

--- a/utils/gtf/gtfcalc.c
+++ b/utils/gtf/gtfcalc.c
@@ -40,6 +40,8 @@
 *
 ****************************************************************************/
 
+#define _XOPEN_SOURCE 600
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
@@ -64,10 +66,12 @@ static GTF_constants GC = {
 
 /*-------------------------- Implementation -------------------------------*/
 
+#ifndef HAVE_ROUND
 static double round(double v)
 {
 	return floor(v + 0.5);
 }
+#endif
 
 static void GetInternalConstants(GTF_constants *c)
 /****************************************************************************
--- a/utils/gtf/scitech.h
+++ b/utils/gtf/scitech.h
@@ -358,6 +358,18 @@
 #define	_ASMAPIP	_ASMAPI *
 #endif
 
+/* We want the _POSIX_VERSION macro if available. */
+#ifdef __UNIX__
+#include <unistd.h>
+#endif
+
+/* Ideally we'd detect if the function is present at build time regardless
+ * of standards, but this is better than nothing. */
+#if (defined(__STDC_VERSION__) && __STDC_VERSION__ >= 199901L) || \
+    (defined(_POSIX_VERSION) &&  _POSIX_VERSION >= 200112L)
+#define HAVE_ROUND
+#endif
+
 /* Useful macros */
 
 #define PRIVATE static
